boards/arm/nrf52840_pca10059/doc/index.rst

```diff
@@ -107,82 +107,189 @@ Programming and Debugging
+   something like ``COMx`` on Windows, and something else on macOS.
+
+   .. code-block:: console
+
+      nrfutil dfu usb-serial -pkg blinky.zip -p /dev/ttyACM0
+
+   When this command exits, observe the green LED on the board blinking,
+   instead of the red LED used by the bootloader.
+
+For more information, see `Nordic Semiconductor USB DFU`_.
+
+Option 2: Using MCUboot in Serial Recovery Mode
+-----------------------------------------------
+
+It is also possible to use the MCUboot bootloader with this board to flash
+Zephyr applications. You need to do some one-time set-up to build and flash
+MCUboot on your board. From that point on, you can build and flash other Zephyr
+applications using MCUboot's serial recovery mode. This process does not
+overwrite the built-in Nordic bootloader, so you can always go back to using
+Option 1 later.
+
+Install `nrfutil`_ and `mcumgr`_ first, and make sure MCUboot's ``imgtool`` is
+available for signing your binary for MCUboot as described on :ref:`west-sign`.
+
+Next, do the **one-time setup** to flash MCUboot. We'll assume you've cloned
+the `MCUboot`_ repository into the directory ``mcuboot``, and that it is next
+to the zephyr repository on your computer.
+
+#. Reset the board into the Nordic bootloader as described above.
+
+#. Compile MCUboot as a Zephyr application.
+
+   .. zephyr-app-commands::
+      :app: mcuboot/boot/zephyr
+      :board: nrf52840_pca10059
+      :build-dir: mcuboot
+      :tool: west
+      :goals: build
+
+#. Package the application for the bootloader using ``nrfutil``:
+
+   .. code-block:: console
+
+      nrfutil pkg generate --hw-version 52 --sd-req=0x00 \
+              --application build/mcuboot/zephyr/zephyr.hex \
+              --application-version 1 mcuboot.zip
+
+#. Flash it onto the board. Note :file:`/dev/ttyACM0` is for Linux; it will be
+   something like ``COMx`` on Windows, and something else on macOS.
+
+   .. code-block:: console
+
+      nrfutil dfu usb-serial -pkg mcuboot.zip -p /dev/ttyACM0
+
+You can now flash a Zephyr application to the board using MCUboot's serial
+recovery mode. We'll use the :ref:`smp_svr_sample` since it's ready to be
+compiled for chain-loading by MCUboot (and itself supports firmware updates
+over Bluetooth).
+
+#. Boot into MCUboot serial recovery mode by plugging the board in with the SW1
+   button pressed down. See above for a picture showing where SW1 is.
+
+   **Do not press RESET**; that will run the Nordic bootloader, which is
+   different than MCUboot.
+
+   A serial port will enumerate on your board. On Windows, "MCUBOOT" should
+   appear under "Other Devices" in the Device Manager (in addition to the usual
+   ``COMx`` device). On Linux, something like
+   :file:`/dev/serial/by-id/usb-ZEPHYR_MCUBOOT_0.01-if00` should be created.
 
-Create an application package, using nrfutil::
+   If no serial port appears, try plugging it in again, making sure SW1 is
```

Comment:
The procedure should work always, without the need to be retried; especially the MCUboot step, there should be no need to retry that.

Reply:
Sure, the code is definitely working; I'm just putting this here for people like me who didn't press SW1 properly and needed to retry. It's concrete troubleshooting advice for if the desired result doesn't happen.